sky/serve/constants.py

```diff
@@ -1,12 +1,14 @@
 """Constants used for SkyServe."""
 
+import os
+
 CONTROLLER_IDLE_MINUTES_TO_AUTOSTOP = 10
 
 CONTROLLER_PREFIX = 'sky-serve-controller-'
 
 CONTROLLER_TEMPLATE = 'sky-serve-controller.yaml.j2'
 
-SERVE_PREFIX = '~/.sky/serve'
+SERVE_PREFIX = os.path.expanduser('~/.sky/serve')
```

Comment:
This variable is also used in generating remote skyserve workdir: https://github.com/skypilot-org/skypilot/blob/3c241b7567bf8d6efa81e999a6e5f52e771d7e41/sky/serve/serve_utils.py#L99-L101
which will later be used on the remove VM, i.e. the username is not the local username, but `gcpuser` or `ubuntu` you normally would see after ssh into your cluster.

Instead, how about we expand it in `sky/serve/__init__.py`? Reference: https://github.com/skypilot-org/skypilot/blob/4fa8aa0e22d76c051793f96c3e7ee2c84cefe2f2/sky/serve/__init__.py#L28